engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+
+/**
+ *
+ */
+public class RenderGraph { // TODO: add extends DirectedAcyclicGraph<Node>
+    private List<Node> nodes;
+
+    public RenderGraph() {
+        nodes = Lists.newArrayList();
+    }
+
+    public String add(Node node, String suggestedId) {
```

Comment:
I've been thinking about the issue of naming these methods a little further. In terms of usage, what is preferable:

`renderGraph.add(shadowMapGeneratorNode, "shadowMapGenerator");`

or 

`renderGraph.addNode(shadowMapGenerator, "shadowMapGenerator")`

Somehow I'm more fond of the second option, because it doesn't require the node instances to have a "Node" suffix to mantain the same degree of closeness to plain english. tdgunes prefers the first option.

@msteiger: I'm sure there is a piece of programming wisdom that can settle this???

The few arguments I can come up with is that there will be more Nodes in the code than .addNode() calls, as such it would be beneficial to reduce the size of node variable names and increase the size of the method name. Also, we are going to have methods such as "addEdge" and I feel it would be more symmetric to have "addNode" methods. I'm also thinking, if somebody writes something like:

renderGraph.add(anUnfamiliarVariable, "anUnfamiliarVariable")

and the code is being reviewed in github, without an IDE at hand to easily find out the variable's type, wouldn't "addNode" be more enlightening?